sdk_contrib/aws_v1_support/src/main/java/io/opentelemetry/sdk/contrib/trace/aws/propagation/AWSXRayPropagator.java

```diff
@@ -0,0 +1,238 @@
+
+  private static final String SAMPLED_FLAG_KEY = "Sampled";
+  private static final int SAMPLED_FLAG_LENGTH = 1;
+  private static final char IS_SAMPLED = '1';
+  private static final char NOT_SAMPLED = '0';
+
+  private static final List<String> FIELDS = Collections.singletonList(TRACE_HEADER_KEY);
+
+  @Override
+  public List<String> fields() {
+    return FIELDS;
+  }
+
+  @Override
+  public <C> void inject(Context context, @Nullable C carrier, Setter<C> setter) {
+    Objects.requireNonNull(context, "context");
+    Objects.requireNonNull(setter, "setter");
+
+    Span span = TracingContextUtils.getSpanWithoutDefault(context);
+    if (span == null) {
+      return;
+    }
+
+    SpanContext spanContext = span.getContext();
+
+    String otTraceId = spanContext.getTraceId().toLowerBase16();
+    String xrayTraceId = TRACE_ID_VERSION + TRACE_ID_DELIMITER +
+        otTraceId.substring(0, TRACE_ID_FIRST_PART_LENGTH) + TRACE_ID_DELIMITER +
+        otTraceId.substring(TRACE_ID_FIRST_PART_LENGTH);
+    String parentId = spanContext.getSpanId().toLowerBase16();
+    char samplingFlag = spanContext.getTraceFlags().isSampled() ? IS_SAMPLED : NOT_SAMPLED;
+
+    String traceHeader = TRACE_ID_KEY + KV_DELIMITER + xrayTraceId + TRACE_HEADER_DELIMITER +
+        PARENT_ID_KEY + KV_DELIMITER + parentId + TRACE_HEADER_DELIMITER +
+        SAMPLED_FLAG_KEY + KV_DELIMITER + samplingFlag;
+    setter.set(carrier, TRACE_HEADER_KEY, traceHeader);
+  }
+
+  @Override
+  public <C> Context extract(Context context, C carrier, Getter<C> getter) {
+    checkNotNull(carrier, "carrier");
+    checkNotNull(getter, "getter");
+
+    SpanContext spanContext = getSpanContextFromHeader(carrier, getter);
+    return TracingContextUtils.withSpan(DefaultSpan.create(spanContext), context);
+  }
+
+  @SuppressWarnings("StringSplitter")
+  private static <C> SpanContext getSpanContextFromHeader(C carrier, Getter<C> getter) {
+    try {
+      String traceHeader = getter.get(carrier, TRACE_HEADER_KEY);
+      if (StringUtils.isNullOrEmpty(traceHeader)) {
+        return SpanContext.getInvalid();
+      }
+
+      String[] parts = traceHeader.split(String.valueOf(TRACE_HEADER_DELIMITER));
+      TraceId traceId = TraceId.getInvalid();
+      SpanId spanId = SpanId.getInvalid();
+      TraceFlags traceFlags = null;
+
+      for (String part : parts) {
+        String[] keyAndValue = extractKeyAndValue(part);
+        if (keyAndValue == null) {
+          continue;
+        }
+        String key = keyAndValue[0];
+        String value = keyAndValue[1];
+
+        switch (key) {
+          case TRACE_ID_KEY:
+            traceId = parseTraceId(value);
+            break;
+          case PARENT_ID_KEY:
+            spanId = parseSpanId(value);
+            break;
+          case SAMPLED_FLAG_KEY:
+            traceFlags = parseTraceFlag(value);
+            break;
+          default:
+            // Ignoring the arbitrary TraceHeader keys in OpenTelemetry
```

Comment:
Think we can put them in tracestate, related to above comment. We should do both or neither though not just one.

Reply:
I'll put a TODO for now